src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,467 @@
+module mpas_geotile_manager
+
+   use iso_c_binding, only : c_float, c_int, c_long, c_loc, c_ptr
+
+   use mpas_constants, only : pii
+
+   use mpas_pool_routines
+   use mpas_stack, only : payload_t
+
+
+   implicit none
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+   private
+
+   public :: tile_t
+   public :: hash_table
+
+   public :: mpas_gen_file
+   public :: mpas_get_tile 
+   public :: mpas_get_tile_neighbors
+   public :: mpas_hash_to_latlon
+   public :: mpas_stack_pop_tile
+
+   type, extends(payload_t) :: tile_t
+      real (c_float), dimension(:,:), pointer, contiguous :: tile
+
+      character (len=StrKIND) :: fname
+      integer :: hash_x
+      integer :: hash_y
+
+      integer, dimension(2) :: x, y ! The tile's range
+      logical :: is_processed = .FALSE.
+   end type tile_t
+
+    type hash_table
+       type(tile_t), pointer :: ptr => null()
+    end type hash_table
+
+   contains
+
+   !***********************************************************************
+   !
+   !  function mpas_get_tile
+   !
+   !> \brief   Retrieve the tile associated with coordinates lat and long
+   !> \author  Miles Curry
+   !> \date    
+   !> \details 
+   !>    This function is an abstraction for reading values from a tile file.
+   !>  It enables the tiles to be read in from disk once and only once.
+   !>
+   !>  It does this by using a data structure of the `tile_t` type. Because
+   !> the we know all about the geographical datafiles from the index file
+   !> we know how many datafiles we know to read to read in, and how large they
+   !> are.
+   !>
+   !> So this function calls mpas_search_tile for a tile which contains `lat` 
+   !> and `lon` and if it is not found to be allocated it calls mpas_add_tile
+   !> to add that tile. If the tile is found, a pointer is returned in tile.
+   !
+   !-----------------------------------------------------------------------
+   subroutine mpas_get_tile(lat, lon, tile, tile_hash, geo_pool)
+
+      implicit none
+
+      ! Input variables 
+      real (kind=RKIND), value :: lat, lon
+      type (tile_t), intent(out), pointer :: tile
+      type (hash_table), pointer, dimension(:,:), intent(inout) :: tile_hash
+      type (mpas_pool_type), intent(inout), pointer :: geo_pool
+
+      tile => null()
+      
+      ! Search for the tile in the data structure (ds) we have implemented, if
+      ! we find the tile in the ds, then the tile_t type will be returned in 
+      ! tile.
+      !
+      ! If we haven't found the tile in the ds, then we need to add the tile
+      ! via add tile, which will also give us the tile_t type that contains
+      ! that tile.
+
+      if (mpas_search_tile(lat, lon, tile, tile_hash, geo_pool)) then
+         ! pass
+      else
+         call mpas_add_tile(lat, lon, tile, tile_hash, geo_pool)
+      endif
```

Comment:
What are your thoughts on rewriting this if-else block to look like:
```
      if (.not. mpas_search_tile(lat, lon, tile, tile_hash, geo_pool)) then
         call mpas_add_tile(lat, lon, tile, tile_hash, geo_pool)
      endif
```
?

Reply:
That sounds like a good change to me! More elegant the better!